pyquil/api/compiler.py

```diff
@@ -0,0 +1,148 @@
+##############################################################################
+# Copyright 2016-2018 Rigetti Computing
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+
+from pyquil.api import Job
+from pyquil.quil import Program
+from ._base_connection import TYPE_MULTISHOT, get_job_id, get_session, \
+    wait_for_job, post_json, get_json
+
+
+class CompilerConnection(object):
+    """
+    Represents a connection to the Quil compiler.
+    """
+
+    def __init__(self, sync_endpoint='https://api.rigetti.com',
+                 async_endpoint='https://job.rigetti.com/beta', api_key=None,
+                 user_id=None, use_queue=False, ping_time=0.1, status_time=2,
+                 default_isa=None):
```

Comment:
Might be worth sync'ing up with @cbosborn about his plan to have a `device` keyword here